src/instruction.rs

```diff
@@ -9,6 +9,7 @@ use crate::err::InterpError;
 pub enum StackValue {
     Integer(u64),
     Boolean(bool),
+    Uninitialised,
```

Comment:
Does this mean `let a;`  is valid? Is it a good idea?

Reply:
`let a;` is a valid statement. Since YAIWR syntax became c-like, having similar behaviour for variable declarations would make sense.
Why would it be a bad idea? Cause its an uninitialised variable?

Reply:
> Why would it be a bad idea? Cause its an uninitialised variable?

One school of thought is: it provides the programmer with a way to crash the program.

The counter argument is: it's best to crash early rather than have a subtle logic error later due to an incorrectly initialised variable.

Up to you :)

Reply:
As in "Null Pointer References: The Billion Dollar Mistake"? Ok, I felt scared. I will disable it!

Reply:
removed :point_right: 37e0ef8fe650bde22c8e04e01d1ac4f035f120e4
I actually didn't have test ffor uninitialised variable declaration :no_mouth:

Reply:
So is `UndeclaredVariable` required now? How can you create one if `let a;` is now invalid?

Reply:
`UndeclaredVariable` will be raised if there is an attempt to assign a value to an undeclared id in scope
Example:
```
a = 1;  // invalid cause a is not declared
```
Valid example:
```
let a = 1;  // valid cause a is declared and assigned
```
Example of variable mutation:
```
let a = 1;
a = 2; // valid cause a is declared
```

Reply:
Sure, but why to you allocate stack space for an uninitialised variable? I don't see a need to store the fact that a scope lookup failed -- just emit an error and bail out. Or did I misunderstand?

Reply:
Ah, I get what you are saying. 
There's no meaning to pushing the `StackValue::Uninitialised` value to stack and it's not used anyway (I think I ended up doing it to satisfy Rust types :no_mouth: )
I can remove the uninitialised variables from the stack space.

Reply:
I think you should be able to delete `StackValue::Uninitialised`

Reply:
removed :point_right: 1cf0d4d88c59846a7346bef2e8274bb62102e34b